syncopy/tests/test_metadata.py

```diff
@@ -455,6 +455,45 @@ def test_sequential_mtmfft(self):
         else:
             print(f"Frequency axes hashes are identical across all {num_hashes_checked} trials.")
 
+    def test_sequential_mtmfft_with_selections(self):
+        """
+        Test metadata propagation with mtmfft in sequential compute mode, with selections.
+
+        This also demonstrates howto use the hashed frequencies from the 2nd return
+        value to ensure they are identical across trials.
+        """
+        # These are known from the input data and cfg.
+        data_size = 100  # Number of samples (per trial) seen by mtmfftm. The full signal returned by _get_fooof_signal() is
+                         # larger, but the cfg.foilim setting (in get_fooof_cfg()) limits to 100 samples.
+
+
+        cfg = TestMetadataUsingMtmfft.get_mtmfft_cfg()
+        cfg.parallel = False
+        cfg.keeptrials = True
+        selected_trials = [3, 5, 7]
+        cfg.select = { 'trials': selected_trials }
+        num_trials_out = len(selected_trials)
+
+        spec_dt = freqanalysis(cfg, self.tfData)
+
+        assert spec_dt.data.shape == (num_trials_out, 1, data_size, 1)
+
+        keys_unique = [kv + "__0_0" for kv in self.expected_metadata_dict_entries]
+
+        assert spec_dt.metadata is not None
+        assert isinstance(spec_dt.metadata, dict)  # Make sure it is a standard dict, not a hdf5 group.
+
+        # Check that the metadata keys are absolute, despite the selection.
+        md_trial_indices = []
+        for k, v in spec_dt.metadata.items():
```

Comment:
The only semantic issue here is, that we have `process_metadata` all over the place, yet here this `.metadata` is more specific (the additional (meta-) returns). Things like `.time`, `.freq.` `.channel` and so on also are called at least implicitly `metadata` atm. On the other hand, this nice demonstration here will go exactly to `process_metadata` as kind of a `finally` block after the CR did its job. I don't think it's a real problem, maybe we find some better/more specific naming in the future. Renaming `process_metadata` to sth like `CR.finalize` (in relation to `CR.initialize`), and re-using the name `process_metadata` to work on what you here call `metadata` I could imagine for example :slightly_smiling_face: